internal/services/dns/dns_zone_resource.go

```diff
@@ -145,6 +147,13 @@ func resourceDnsZone() *pluginsdk.Resource {
 
 			"tags": commonschema.Tags(),
 		},
+		// CustomizeDiff: pluginsdk.CustomDiffWithAll(
+		// 	pluginsdk.ValueChangeConditionFunc()
+		// 	pluginsdk.ForceNewIfChange("host_name", func(ctx context.Context, old, new, meta interface{}) bool {
+		// 		return false
+		// 		//return (old != nil && old.(string) != "") && (new == nil || new.(string) == "")
+		// 	}),
+		// ),
```

Comment:
From the Azure document and the Portal behavior, the `host_name` doesn't want to be changed. So shall we keep this and mention that if this property is changed, the resource will be recreated?

Otherwise, if this is not needed, please just remove it.

Reply:
this was an oversight, I removed it.